### DISCUSSION TITLE
How to add a unsupported browser message

### DISCUSSION THREAD

**Discussion by vidyashreen1991** [vercel/next.js#28443] [Help]
Where exactly to add the browser check to show the message

**Comment by oliverloops** [comment]
Hi @vidyashreen1991, for this you can use one of the built in browser objects called **_User Agent_** and implement it using just plain JavaScript, the idea behind this object is to detect through one of their methods in which browser you are, as an example: 

```
if (navigator.userAgent.indefOf("Chrome" !== -1)){
   console.log("You're using Chrome")

} else if (navigator.userAgent.indefOf("Safari" !== -1)){
   console.log("You're using Safari")

} else if (navigator.userAgent.indefOf("Firefox" !== -1)) {
   console.log("You're using Firefox")

} else {
    console.log("Any other browser")
}
```

But you should take some considerations here, is recommended to use only as a way to provide warning or error messages but it's not intended for testing supported features or for blocking specific features in some browser. You can read more about this on the MDN official site: https://developer.mozilla.org/en-US/docs/Web/HTTP/Browser_detection_using_the_user_agent

Note: In some references `navigator.userAgent.indefOf()` method is used through `window` object as `window.navigator.userAgent.indefOf()`, if you're combining this with Next.js don't forget to wrap this objects calls in something like 

```
if (typeof window !== undefined) {
  //Your code
}
```
To avoid to run this code on the server side and reduce any conflict.

I'll hope this helps, let me know
